airflow/contrib/example_dags/example_spark_kubernetes_operator.py

```diff
@@ -0,0 +1,128 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+
+import yaml
+
+# [START import_module]
+# The DAG object; we'll need this to instantiate a DAG
+from airflow import DAG
+# Operators; we need this to operate!
+from airflow.contrib.operators.spark_kubernetes_operator import SparkKubernetesOperator
+from airflow.contrib.sensors.spark_kubernetes_sensor import SparkKubernetesSensor
+from airflow.utils.dates import days_ago
+
+# [END import_module]
+
+# [START default_args]
+# These args will get passed on to each operator
+# You can override them on a per-task basis during operator initialization
+default_args = {
+    'owner': 'airflow',
+    'depends_on_past': False,
+    'start_date': days_ago(1),
+    'email': ['airflow@example.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'max_active_runs': 1
+}
+# [END default_args]
+
+# [START instantiate_dag]
+spark_application_yaml = """
+#
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: spark-pi
+  namespace: default
+spec:
+  type: Scala
+  mode: cluster
+  image: "gcr.io/spark-operator/spark:v2.4.4"
+  imagePullPolicy: Always
+  mainClass: org.apache.spark.examples.SparkPi
+  mainApplicationFile: "local:///opt/spark/examples/jars/spark-examples_2.11-2.4.4.jar"
+  sparkVersion: "2.4.4"
+  restartPolicy:
+    type: Never
+  volumes:
+    - name: "test-volume"
+      hostPath:
+        path: "/tmp"
+        type: Directory
+  driver:
+    cores: 1
+    coreLimit: "1200m"
+    memory: "512m"
+    labels:
+      version: 2.4.4
+    serviceAccount: default
+    volumeMounts:
+      - name: "test-volume"
+        mountPath: "/tmp"
+  executor:
+    cores: 1
+    instances: 1
+    memory: "512m"
+    labels:
+      version: 2.4.4
+    volumeMounts:
+      - name: "test-volume"
+        mountPath: "/tmp"
+"""
```

Comment:
Can you store content inthe file and `template_ext`, please? DAG should be simple. 
https://airflow.readthedocs.io/en/latest/_api/airflow/models/baseoperator/index.html#airflow.models.baseoperator.BaseOperator.template_ext

Reply:
I saw the other Kubernetes operators put dict with Kubernetes object to be applied by the operator.
if I put the contents in a file where to put it?

Reply:
The API of KubernetesPodOperator is being modernized.  
https://github.com/apache/airflow/pull/6375
https://github.com/apache/airflow/pull/6230
You should create this file in the same directory as the python file - `example_dag`

Reply:
Mixing code from different technologies is very problematic.I remember there is a word that describes this problem, but I don't remember the exact name. A similar problem occurs with SQL and other languages. SQL can be used in other code, but when it is very complex people started using ORM to connect the two worlds - SQL and Objects. 

You can create a file in the directory next to the file. For example. If your Python file is in the ``airflow/contrib/example_dags`` directory, you can keep additional files in the ``airflow/contrib/example_dags`` directory

Reply:
fixed it: user will provide the path for the sparkApplication yaml I put a comment with a link to spark-pi example on spark-on-k8s git